Source/items.cpp

```diff
@@ -4130,29 +4142,9 @@ void DrawUniqueInfo(const Surface &out)
 	rect.position.y += (12 - uitem.UINumPL) * 12;
 	PrintItemPower(uitem.powers[0].type, &curruitem);
 	DrawString(out, tempstr, rect, UIS_SILVER | UIS_CENTER);
-	if (uitem.UINumPL > 1) {
-		rect.position.y += 2 * 12;
-		PrintItemPower(uitem.powers[1].type, &curruitem);
-		DrawString(out, tempstr, rect, UIS_SILVER | UIS_CENTER);
-	}
-	if (uitem.UINumPL > 2) {
-		rect.position.y += 2 * 12;
-		PrintItemPower(uitem.powers[2].type, &curruitem);
-		DrawString(out, tempstr, rect, UIS_SILVER | UIS_CENTER);
-	}
-	if (uitem.UINumPL > 3) {
-		rect.position.y += 2 * 12;
-		PrintItemPower(uitem.powers[3].type, &curruitem);
-		DrawString(out, tempstr, rect, UIS_SILVER | UIS_CENTER);
-	}
-	if (uitem.UINumPL > 4) {
-		rect.position.y += 2 * 12;
-		PrintItemPower(uitem.powers[4].type, &curruitem);
-		DrawString(out, tempstr, rect, UIS_SILVER | UIS_CENTER);
-	}
-	if (uitem.UINumPL > 5) {
+	for (int i = 1; i < uitem.UINumPL; i++) {

```

Comment:
This loop can start from 0, line 4142 can be split to make the first bonus behave the same as subsequent bonuses.

Reply:
Could potentially add an assert to document that UINumPL is at most 6.

Reply:
Would having bool first = true; be an ok way to handle the extra offsets needed for the first power?

Reply:
I was thinking something like:

```cpp
rect.position.y += (10 - uitem.UINumPL) * 12; // using 10 - power level instead of 12
for (int i = 0; i < uitem.UINumPL; i++) {
    rect.position.y += 2 * 12; // this allows treating the first case as normal and gets back to the expected offset
    PrintItemPower();
    DrawString();
}
```

Could also do the same thing by leaving the first increment as 12 - power and moving the inner increment to the end of the loop.